File: portal/core/src/main/java/com/serious/portal/HelpAdministrationService.kt
package com.serious.portal
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import com.serious.service.Service
import com.serious.service.ServiceInterface
import org.springframework.web.bind.annotation.*

@ServiceInterface
@RequestMapping("help-administration/")
@RestController
interface HelpAdministrationService : Service {
    @PostMapping("save-help/{feature}")
    fun saveHelp(@PathVariable feature : String, @RequestBody help : String)

    @GetMapping("read-help/{feature}")
    fun readHelp(@PathVariable feature : String) : String

    @GetMapping("delete-help/{feature}")
    fun deleteHelp(@PathVariable feature : String)

    @GetMapping("read-entries/")
    fun readEntries() : List<String>
}

File: modulefederation/libs/portal/src/lib/help/help-administration.service.ts
/**
 * @COPYRIGHT (C) 2023 Andreas Ernst
 * don't touch!
 * generated at Wed Dec 27 11:23:47 CET 2023 with typescript V1.0
 */

import { Injectable, Injector } from "@angular/core"
import { Observable } from "rxjs"
import { AbstractHTTPService, Service } from "../common/communication"

@Injectable({providedIn: 'root'})
@Service({domain: "admin", prefix: "/help-administration/"})
export class HelpAdministrationService extends AbstractHTTPService {
	// constructor

	constructor(injector: Injector) {
		super(injector)
	}

	// public methods

	public readEntries() : Observable<string[]> {
		return this.get<string[]>(`read-entries/`)
	}

	public saveHelp(feature : string, help : string) : Observable<any> {
		return this.post<string>(`save-help/${feature}`, help)
	}

	public deleteHelp(feature : string) : Observable<any> {
		return this.get<string>(`delete-help/${feature}`)
	}

	public readHelp(feature : string) : Observable<string> {
  		return this.get<string>(`read-help/${feature}`)
  	}
}
